pkg/provisioner/ironic/ironic.go

```diff
@@ -354,6 +354,39 @@ func (p *ironicProvisioner) findExistingHost() (ironicNode *nodes.Node, err erro
 
 }
 
+func (p *ironicProvisioner) updateAutomatedClean(ironicNode *nodes.Node) (err error) {
+	p.log.Info("setting cleaning mode",
+		"ID", ironicNode.UUID,
+		"mode", p.host.Spec.AutomatedCleaningMode)
+
+	p.publisher("CleaningMode", string(p.host.Spec.AutomatedCleaningMode))
+
+	value := p.host.Spec.AutomatedCleaningMode != metal3v1alpha1.CleaningModeDisabled
+
+	updates := nodes.UpdateOpts{
+		nodes.UpdateOperation{
+			Op:    nodes.ReplaceOp,
+			Path:  "/automated_clean",
+			Value: value,
+		},
+	}
+
+	_, err = nodes.Update(p.client, ironicNode.UUID, updates).Extract()
+
+	switch err.(type) {
+	case nil:
+	case gophercloud.ErrDefault409:
+		p.log.Info("could not update ironic node automated_clean, busy")
+		_, err = retryAfterDelay(provisionRequeueDelay)
+		return
+	default:
+		_, err = transientError(errors.Wrap(err, "failed to update ironic node automated_clean"))
```

Comment:
Using the transientError call here is pointless in a function that only returns an `error`.

Reply:
True, will fix it once we decide on the questions above.